### PR TITLE
🚀 Add VAD dataset

### PR DESCRIPTION
**Pull request by abc-125**
## 📝 Description

Added one-class (unsupervised) version of [VAD dataset](https://github.com/abc-125/vad), which contains one category of complex industrial objects.

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [x] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [x] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/data/datamodules/image/__init__.py [hunk 1]
```diff
@@ -55,15 +58,16 @@ class ImageDataFormat(str, Enum):
...
     "BTech",
     "Datumaro",
     "Folder",
     "Kolektor",
     "MVTecAD",
     "MVTec",  # Include both for backward compatibility
+    "VAD",
     "Visa",
-]
+]
```
Comment (samet-akcay) [inline src/anomalib/data/datamodules/image/__init__.py:73]:
```suggestion
]

```

File: src/anomalib/data/datamodules/image/vad.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    https://creativecommons.org/licenses/by-nc-sa/4.0/
+
+Reference:
+    Aimira Baitieva, David Hurych, Victor Besnier, Olivier Bernard:
+    Supervised Anomaly Detection for Complex Industrial Images; in:
+    The IEEE/CVF Conference on Computer Vision and Pattern Recognition, 2024,
+    pp. 17754-17762, DOI: 10.1109/CVPR52733.2024.01681.
+"""
+
+# Copyright (C) 2022-2025 Intel Corporation
```
Comment (samet-akcay) [inline src/anomalib/data/datamodules/image/vad.py:37]:
```suggestion
# Copyright (C) 2025 Intel Corporation
```

File: src/anomalib/data/datamodules/image/vad.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            Defaults to ``ValSplitMode.SAME_AS_TEST``.
+        val_split_ratio (float): Fraction of data to use for validation.
+            Defaults to ``0.5``.
+        seed (int | None, optional): Seed for reproducibility.
+            Defaults to ``None``.
+
+    Example:
+        Create VAD datamodule with default settings::
+
+            >>> datamodule = VAD()
```
Comment (samet-akcay) [inline src/anomalib/data/datamodules/image/vad.py:94]: Do you have built-in validation set?

File: tests/unit/data/datamodule/image/test_vad.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+"""Unit Tests - VAD Datamodule."""
+
+# Copyright (C) 2023-2025 Intel Corporation
```
Comment (samet-akcay) [inline tests/unit/data/datamodule/image/test_vad.py:3]:
```suggestion
# Copyright (C) 2025 Intel Corporation
```

File: src/anomalib/data/datamodules/image/vad.py (hunk 3 above)
Comment (samet-akcay) [reply src/anomalib/data/datamodules/image/vad.py:94]: If we support supervised anomaly detection models, we could add VAD's full supervised version. In this case, how much change/modification do you think we need to add to this implementation?

Comment (abc-125) [reply src/anomalib/data/datamodules/image/vad.py:94]: Not sure, it would depend on the implementation, but it should not be too complicated. 

I would imagine having three types of supervised AD datasets; one is adding a set of bad images for training (VAD), another is having several splits over the entire dataset which contains good and bad images into training and test (KolektorSDD), and last one is taking N images from bad images from the test set randomly for existing not supervised datasets (supervised version of MVTecAD, etc.).

**Review by samet-akcay — APPROVED** [review]
Thanks for this great contribution!
